pkg/virtctl/vnc/vnc.go

```diff
@@ -165,78 +169,21 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		listenResChan <- err
 	}()
 
-	// execute VNC
-	go func() {
-		defer close(doneChan)
-		port := ln.Addr().(*net.TCPAddr).Port
-		args := []string{}
-
-		vncBin := ""
-		osType := runtime.GOOS
-		switch osType {
-		case "darwin":
-			if matches, err := filepath.Glob(MACOS_TIGER_VNC_PATTERN); err == nil && len(matches) > 0 {
-				// Always use the latest version
-				vncBin = matches[len(matches)-1]
-				args = tigerVncArgs(port)
-			} else if err == filepath.ErrBadPattern {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(MACOS_CHICKEN_VNC); err == nil {
-				vncBin = MACOS_CHICKEN_VNC
-				args = chickenVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(MACOS_REAL_VNC); err == nil {
-				vncBin = MACOS_REAL_VNC
-				args = realVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := exec.LookPath(REMOTE_VIEWER); err == nil {
-				// fall back to user supplied script/binary in path
-				vncBin = REMOTE_VIEWER
-				args = remoteViewerArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			}
-		case "linux", "windows":
-			if _, err := exec.LookPath(REMOTE_VIEWER); err == nil {
-				vncBin = REMOTE_VIEWER
-				args = remoteViewerArgs(port)
-			} else if _, err := exec.LookPath(TIGER_VNC); err == nil {
-				vncBin = TIGER_VNC
-				args = tigerVncArgs(port)
-			} else {
-				viewResChan <- fmt.Errorf("could not find %s or %s binary in $PATH",
-					REMOTE_VIEWER, TIGER_VNC)
-				viewResChan <- err
-				return
-			}
-		default:
-			viewResChan <- fmt.Errorf("virtctl does not support VNC on %v", osType)
-			return
-		}
+	port := ln.Addr().(*net.TCPAddr).Port
 
-		if vncBin == "" {
-			glog.Errorf("No supported VNC app found in %s", osType)
-			err = fmt.Errorf("No supported VNC app found in %s", osType)
-		} else {
-			if glog.V(4) {
-				glog.Infof("Executing commandline: '%s %v'", vncBin, args)
-			}
-			cmnd := exec.Command(vncBin, args...)
-			output, err := cmnd.CombinedOutput()
-			if err != nil {
-				glog.Errorf("%s execution failed: %v, output: %v", vncBin, err, string(output))
-			} else {
-				glog.V(2).Infof("remote-viewer output: %v", string(output))
-			}
+	if proxyOnly {
+		defer close(doneChan)
+		optionString, err := json.Marshal(struct {
+			Port int `json:"port"`
+		}{port})
+		if err != nil {
+			return fmt.Errorf("Error encountered: %s", err.Error())
 		}
-		viewResChan <- err
-	}()
+		fmt.Println(string(optionString))
```

Comment:
Can you replace this line with `fmt.Fprintln(cmd.OutOrStdout(), string(optionString))`?

Reply:
Done